### PR TITLE
[FEATURE] Improve estimation histogram computation in NumericMetricRangeMultiBatchParameterBuilder to include both counts and bin edges

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Currently, `estimation_histogram` field of `NumericRangeEstimationResult` stores only counts.  However, having also the bin edges is useful, because the latter are the actual values of the quantities being counted.  This work insures that both are stored properly.  Even though `estimation_histogram` is disabled by default, it will more useful when enabled.
* Fix errors in existing tests (they were not checking counts of metric values correctly).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1133
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/helpers/util.py [hunk 1]
```diff
@@ -484,8 +484,22 @@ def compute_quantiles(
...
+            (
+                np.pad(
+                    array=histogram[0],
+                    pad_width=(0, 1),
+                    mode="constant",
+                    constant_values=0,
+                ),
+                histogram[1],
+            )
+        ),
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/helpers/util.py]: ❤️

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/helpers/util.py]: This taught me about `np.pad`!

File: great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py [hunk 2]
```diff
@@ -828,15 +847,27 @@ def _get_oneshot_estimate(
...
+                (
+                    np.pad(
+                        array=histogram[0],
+                        pad_width=(0, 1),
+                        mode="constant",
+                        constant_values=0,
+                    ),
+                    histogram[1],
+                )
+            ),
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py]: I see this same code block used 5 times. Can we refactor this out into its own method?

File: great_expectations/rule_based_profiler/helpers/util.py [hunk 3]
```diff
@@ -484,8 +484,22 @@ def compute_quantiles(
         axis=0,
         method=quantile_statistic_interpolation_method,
     )
+
+    histogram: Tuple[np.ndarray, np.ndarray] = np.histogram(
+        a=metric_values, bins=NUM_HISTOGRAM_BINS
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/helpers/util.py]: Are we planning to allow `NUM_HISTOGRAM_BINS` to be set by the user?

**Review by NathanFarmer — COMMENTED** [review]
Looks good! A few comments from me.

File: great_expectations/rule_based_profiler/helpers/util.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/helpers/util.py]: @NathanFarmer Right now, there does not seem to be a need for it to be customized.  We can always do it in the future, if called for.  Thanks!

**Review by NathanFarmer — APPROVED** [review]
Looks good to me!

**Pull request merged**